conans/client/installer.py

```diff
@@ -509,6 +502,8 @@ def _handle_node_cache(self, node, keep_build, processed_package_references, rem
                     log_package_got_from_local_cache(pref)
                     self._recorder.package_fetched_from_cache(pref)
 
+            package_folder = layout.package(pref)
+            assert os.path.isdir(package_folder), "Package folder must exist %s" % package_folder
```

Comment:
Add more info to the message: if it is using short-paths here we will get only an obfuscated path. Add the `pref` itself.